#! /bin/sh /usr/share/dpatch/dpatch-run
## 18_signalhandler-fix.dpatch by Tobias Grimm <tg@e-tobi.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: This patch fixes a problem with calling isyslog from within the
## DP: signal handler. As suggested by Udo Richter, this patch reverts
## DP: changes done in a previous upstream update.

@DPATCH@
diff -urNad vdr-1.6.0~/vdr.c vdr-1.6.0/vdr.c
--- vdr-1.6.0~/vdr.c	2008-03-14 14:22:39.000000000 +0100
+++ vdr-1.6.0/vdr.c	2008-07-24 00:37:31.000000000 +0200
@@ -141,7 +141,6 @@
 
 static void SignalHandler(int signum)
 {
-  isyslog("caught signal %d", signum);
   switch (signum) {
     case SIGPIPE:
          break;
@@ -1255,6 +1254,8 @@
         PluginManager.MainThreadHook();
         }
 
+  if (LastSignal)
+     isyslog("caught signal %d", LastSignal);
   if (ShutdownHandler.EmergencyExitRequested())
      esyslog("emergency exit requested - shutting down");
 
